instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,141 @@
+            end
+          end
+
+          # Handles the endpoint_run_filters.grape events
+          def endpoint_run_filters(_name, start, finish, _id, payload)
+            filters = payload[:filters]
+            type = payload[:type]
+
+            # Prevent submitting empty filters
+            return if (!filters || filters.empty?) || !type || (finish - start).zero?
+
+            name = span_name(payload[:endpoint])
+            attributes = {
+              'component' => 'web',
+              'operation' => 'endpoint_run_filters',
+              'grape.filter.type' => type.to_s
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          # Handles the format_response.grape event
+          def format_response(_name, start, _finish, _id, payload)
+            endpoint = payload[:env]['api.endpoint']
+            name = span_name(endpoint)
+            attributes = {
+              'component' => 'template',
+              'operation' => 'format_response',
+              'grape.formatter.type' => formatter_type(payload[:formatter])
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          private
+
+          def tracer
+            Grape::Instrumentation.instance.tracer
+          end
+
+          def span_name(endpoint)
+            "#{api_instance(endpoint)} #{request_method(endpoint)} #{path(endpoint)}"
+          end
+
+          def run_attributes(payload)
+            endpoint = payload[:endpoint]
+            path = path(endpoint)
+            {
+              'component' => 'web',
+              'operation' => 'endpoint_run',
+              'grape.route.endpoint' => api_instance(endpoint),
+              'grape.route.path' => path,
+              'grape.route.method' => endpoint.options[:method].first,
+              OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => request_method(endpoint),
+              OpenTelemetry::SemanticConventions::Trace::HTTP_ROUTE => path
+            }
+          end
+
+          def handle_error(span, exception)
+            span.record_exception(exception)
+            span.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{exception.class}")
+            return unless exception.respond_to?('status') && exception.status
+
+            span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, exception.status)
+          end
+
+          def api_instance(endpoint)
+            endpoint.options[:for].base.to_s
+          end
+
+          def request_method(endpoint)
+            endpoint.options.fetch(:method).first
```

Comment:
can this throw, should we handle that ? (fetch fails ?)

Reply:
```suggestion
            endpoint.options.fetch(:method, []).first
```
☝️ will always succeed, and should return `nil` if `{}`, or `{ method: [] }`.

Reply:
The `fetch` was leftover from copypasta 😅 🙈 
I've removed it and added safe navigation 😄